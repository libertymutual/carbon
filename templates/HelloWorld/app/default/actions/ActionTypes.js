// APP ACTIONS

// AUTH ACTIONS
export const LOGIN = 'login';
export const LOGOUT = 'logout';
export const AUTH_UPDATE_AUTH_URL = 'auth_update_auth_url'
export const AUTH_UPDATE_CLIENT_ID = 'auth_update_client_id'
export const AUTH_UPDATE_CLIENT_SECRET = 'auth_update_client_secret'
export const AUTH_UPDATE_CONFIG_NAME = 'auth_update_config_name'
export const AUTH_UPDATE_PASSWORD = 'auth_update_password'
export const AUTH_UPDATE_USERNAME = 'auth_update_username'
export const AUTH_UPDATE_VALIDATOR_ID = 'auth_update_validator_id'

// COUNTER ACTIONS
export const INCREMENT = 'increment';
export const DECREMENT = 'decrement';

// BIOMETRIC ACTIONS
export const UPDATE_USERNAME = 'update_username';
export const UPDATE_PASSWORD = 'update_password';
export const SAVE_CREDENTIALS = 'save_credentials';
export const DELETE_CREDENTIALS = 'delete_credentials';
