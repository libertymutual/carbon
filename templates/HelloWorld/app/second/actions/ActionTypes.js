// APP ACTIONS

// HELLO WORLD ACTIONS
export const HELLO = 'hello';
export const GOODBYE = 'goodbye';

// COUNTER ACTIONS
export const INCREMENT = 'increment';
export const DECREMENT = 'decrement';
