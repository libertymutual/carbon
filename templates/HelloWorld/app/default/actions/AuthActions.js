import * as ActionTypes from '../actions/ActionTypes';
import { NativeModules } from 'react-native';

const { Auth } = NativeModules;

export type AuthActionsType = {
  login: () => ((Object) => void, () => Object) => {},
logout: () => () => { },
};


/**
 * Auth related action creators
 */
export default {

  /**
   * Logs the user in
   * @returns {function} A function which dispatches a LOGIN action
   */
  login() {
    return (dispatch: Object => void, getState: () => Object) => {
      const username = getState().username;
      const password = getState().password;
      Auth.login(username, password).then((isLoggedIn) => {
        if (isLoggedIn) {
          dispatch({ type: ActionTypes.LOGIN })
        }
      })
    }
  },

  /**
   * Logs the user out
   * @returns {function} A function which dispatches a LOGOUT action
   */
  logout() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.LOGOUT });
    };
  },

  updateAuthUrl(authUrl) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_AUTH_URL, authUrl })
    }
  },

  updateClientId(clientId) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CLIENT_ID, clientId })
    }
  },

  updateClientSecret(clientSecret) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CLIENT_SECRET, clientSecret })
    }
  },

  updateConfigName(configName) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_CONFIG_NAME, configName })
    }
  },

  updatePassword(password) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_PASSWORD, password })
    }
  },

  updateUsername(username) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_USERNAME, username })
    }
  },

  updateValidatorId(validatorId) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.AUTH_UPDATE_VALIDATOR_ID, validatorId })
    }
  },
};
