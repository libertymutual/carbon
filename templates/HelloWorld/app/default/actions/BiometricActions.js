  import * as ActionTypes from '../actions/ActionTypes';

/**
 * Counter related action creators
 */
export default {

  /**
   * Updates the username value
   * @returns {function} A function which dispatches an UPDATE_USERNAME action
   */
  updateUsername(username) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.UPDATE_USERNAME, username });
    };
  },

  /**
   * Updates the password value
   * @returns {function} A function which dispatches an UPDATE_PASSWORD action
   */
  updatePassword(password) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.UPDATE_PASSWORD, password });
    };
  },

  /**
   * Updates state to reflect that credentials have been saved
   * @returns {function} A function which dispatches a SAVE_CREDENTIALS action
   */
  saveCredentials() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.SAVE_CREDENTIALS });
    };
  },

  /**
   * Updates state to reflect that credentials have been saved
   * @returns {function} A function which dispatches a DELETE_CREDENTIALS action
   */
  deleteCredentials() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.DELETE_CREDENTIALS });
    };
  },
};
