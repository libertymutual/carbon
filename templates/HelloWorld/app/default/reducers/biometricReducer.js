import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
    username: "TestUsername",
    password: "TestPassword",
    credentialsSaved: false
};

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function biometricReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.UPDATE_PASSWORD:
      return { ...state, password: action.password };  
    case ActionTypes.SAVE_CREDENTIALS:
      return { ...state, credentialsSaved: true };
    case ActionTypes.DELETE_CREDENTIALS:
      return { ...state, credentialsSaved: false };
    default:
      return state;
  }
}
