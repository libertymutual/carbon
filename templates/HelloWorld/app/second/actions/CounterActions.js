import * as ActionTypes from '../actions/ActionTypes';


/**
 * Counter related action creators
 */
export default {

  /**
   * Increments the counter
   * @returns {function} A function which dispatches an INCREMENT action
   */
  increment() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.INCREMENT });
    };
  },

  /**
   * Decrements the counter
   * @returns {function} A function which dispatches a DECREMENT action
   */
  decrement() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.DECREMENT });
    };
  },
};
