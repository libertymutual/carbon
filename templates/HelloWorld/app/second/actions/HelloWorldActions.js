import * as ActionTypes from '../actions/ActionTypes';

/**
 * Hello World related action creators
 */
export default {

  /**
   * Says hello world
   * @returns {function} A function which dispatches a HELLO action
   */
  hello() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.HELLO });
    };
  },

  /**
   * Says goodbye world
   * @returns {function} A function which dispatches a GOODBYE action
   */
  goodbye() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.GOODBYE });
    };
  },
};
