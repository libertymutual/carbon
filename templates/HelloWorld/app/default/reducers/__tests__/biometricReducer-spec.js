import 'react-native';
import React from 'react';
import biometricReducer from '../biometricReducer';
import { initialState } from '../biometricReducer';
import * as ActionTypes from '../../actions/ActionTypes';

describe('biometricReducer tests', () => {
  test('test update username', () => {
    expect(biometricReducer(initialState, { type: ActionTypes.UPDATE_USERNAME, username: "NewUsername" })).toMatchSnapshot();
  });

  test('test update password', () => {
    expect(biometricReducer(initialState, { type: ActionTypes.UPDATE_PASSWORD, password: "NewPassword" })).toMatchSnapshot();
  });

  test('test save credentials', () => {
    expect(biometricReducer(initialState, { type: ActionTypes.SAVE_CREDENTIALS })).toMatchSnapshot();
  });

  test('test delete credentials', () => {
    expect(biometricReducer(initialState, { type: ActionTypes.DELETE_CREDENTIALS })).toMatchSnapshot();
  });
});
