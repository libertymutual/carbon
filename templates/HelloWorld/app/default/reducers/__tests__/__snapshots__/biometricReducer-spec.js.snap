// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`biometricReducer tests test delete credentials 1`] = `
Object {
  "credentialsSaved": false,
  "password": "TestPassword",
  "username": "TestUsername",
}
`;

exports[`biometricReducer tests test save credentials 1`] = `
Object {
  "credentialsSaved": true,
  "password": "TestPassword",
  "username": "TestUsername",
}
`;

exports[`biometricReducer tests test update password 1`] = `
Object {
  "credentialsSaved": false,
  "password": "NewPassword",
  "username": "TestUsername",
}
`;

exports[`biometricReducer tests test update username 1`] = `
Object {
  "credentialsSaved": false,
  "password": "TestPassword",
  "username": "NewUsername",
}
`;
